pkg/controller/installation/products/fuse/reconciler.go

```diff
@@ -0,0 +1,212 @@
+		return v1alpha1.PhaseFailed, errors.Wrap(err, " failed to reconcile subscription for fuse ")
+	}
+	reconciledPhase, err = r.reconcileCustomResource(ctx,inst)
+	if err != nil{
+		return v1alpha1.PhaseFailed, errors.Wrap(err, " failed to reconcile fuse custom resource ")
+	}
+
+	logrus.Infof("End of reconcile Phase : ", reconciledPhase)
+
+	return v1alpha1.StatusPhase(v1alpha1.PhaseCompleted), nil
+}
+
+func (r *Reconciler) reconcileNamespace(ctx context.Context, inst *v1alpha1.Installation) (v1alpha1.StatusPhase, error) {
+	if !inst.Spec.CreateNamespaces {
+		r.logger.Infof("skipping creating namespace: %s", r.Config.GetNamespace())
+		return v1alpha1.PhaseCreatingSubscription, nil
+	}
+	ns := &v1.Namespace{
+		ObjectMeta: v12.ObjectMeta{
+			Name: r.Config.GetNamespace(),
+		},
+	}
+	ns, err := r.nsReconciler.Reconcile(ctx, ns, inst)
+	if err != nil {
+		return v1alpha1.PhaseFailed, errors.Wrap(err, "failed to reconcile fuse namespace "+r.Config.GetNamespace())
+	}
+	if ns.Status.Phase == v1.NamespaceTerminating {
+		r.logger.Debugf("namespace %s is terminating, maintaining phase to try again on next reconcile", r.Config.GetNamespace())
+		return v1alpha1.PhaseAwaitingNS, nil
+	}
+	if ns.Status.Phase != v1.NamespaceActive {
+		return v1alpha1.PhaseAwaitingNS, nil
+	}
+	// all good return no status if already
+	return v1alpha1.PhaseNone, nil
+}
+
+func (r *Reconciler) reconcileSubscription(ctx context.Context) (v1alpha1.StatusPhase, error) {
+	// NEED to ensure a subscription is created if not exists
+	// need to make sure there is only one operator source
+	r.logger.Infof("reconciling subscription %s from channel %s in namespace: %s", defaultSubscriptionName, "integreatly", r.Config.GetNamespace())
+	err := r.mpm.CreateSubscription(
+		marketplace.GetOperatorSources().Integreatly,
+		r.Config.GetNamespace(),
+		defaultSubscriptionName,
+		marketplace.IntegreatlyChannel,
+		[]string{r.Config.GetNamespace()},
+		v1alpha12.ApprovalAutomatic)
+	if err != nil && !errors2.IsAlreadyExists(err) {
+		return v1alpha1.PhaseFailed, errors.Wrap(err, fmt.Sprintf("could not create subscription in namespace: %s", r.Config.GetNamespace()))
+	}
+	return r.handleAwaitingOperator(ctx)
+}
+
+func (r *Reconciler)reconcileCustomResource(ctx context.Context, install *v1alpha1.Installation)(v1alpha1.StatusPhase, error)  {
+	ref := v12.NewControllerRef(install,v1alpha1.SchemaGroupVersionKind)
+	intLimit := -1
+	cr := syn.Syndesis{
+		ObjectMeta: v12.ObjectMeta{
+			OwnerReferences:[]v12.OwnerReference{
+				*ref,
+			},
+			Namespace:r.Config.GetNamespace(),
+			Name:"integreatly",
+		},
+		TypeMeta: v12.TypeMeta{},
+		Spec:syn.SyndesisSpec{
+			Integration:syn.IntegrationSpec{
+				Limit: &intLimit,
+			},
+			Components:syn.ComponentsSpec{
+				Server:syn.ServerConfiguration{
+					Features:syn.ServerFeatures{
+						ExposeVia3Scale:true,
+					},
+				},
+			},
+		},
+	}
+	if err := r.client.Get(ctx,pkgclient.ObjectKey{Name: cr.Name},&cr); err != nil{
```

Comment:
Can't we just try to create and ignore an `IsAlreadyExists` error, so we only have to connect to the API once?